pkg/soc/soc.go

```diff
@@ -22,130 +22,131 @@ const (
 	minChunkSize  = IdSize + SignatureSize + swarm.SpanSize
 )
 
-// Id is a soc identifier
-type Id []byte
+var (
+	errNoOwner        = errors.New("soc: owner not set")
+	errWrongChunkSize = errors.New("soc: chunk length is less than minimum")
+)
 
-// Owner is a wrapper that enforces valid length address of soc owner.
-type Owner struct {
-	address []byte
-}
+// ID is a soc identifier
+type ID []byte
+
+// Owner is the address in bytes of soc owner.
+type Owner []byte
 
-// NewOwner creates a new Owner.
-func NewOwner(address []byte) (*Owner, error) {
+// NewOwner ensures a valid length address of soc owner.
+func NewOwner(address []byte) (Owner, error) {
 	if len(address) != crypto.AddressSize {
-		return nil, fmt.Errorf("invalid address %x", address)
+		return nil, fmt.Errorf("soc: invalid address %x", address)
 	}
-	return &Owner{
-		address: address,
-	}, nil
+	return address, nil
 }
 
-// Soc wraps a single soc.
+// Soc wraps a single-owner chunk.
 type Soc struct {
-	id        Id
+	id        ID
+	owner     Owner
 	signature []byte
-	signer    crypto.Signer
-	owner     *Owner
-	Chunk     swarm.Chunk
+	chunk     swarm.Chunk // wrapped chunk.
 }
 
-// NewChunk is a convenience function to create a single-owner chunk ready to be sent
-// on the network.
-func NewChunk(id Id, ch swarm.Chunk, signer crypto.Signer) (swarm.Chunk, error) {
-	s := New(id, ch)
-	err := s.AddSigner(signer)
-	if err != nil {
-		return nil, err
+// New creates a new Soc representation from arbitrary soc id and
+// a content-addressed chunk.
+func New(id ID, ch swarm.Chunk) *Soc {
+	return &Soc{
+		id:    id,
+		chunk: ch,
 	}
-	return s.ToChunk()
 }
 
-// NewChunk is a convenience function to create a single-owner chunk ready to be sent
-// on the network.
-func NewSignedChunk(id Id, ch swarm.Chunk, owner, sig []byte) (swarm.Chunk, error) {
+// NewSignedSoc creates a single-owner chunk based on already signed data.
+func NewSigned(id ID, ch swarm.Chunk, owner, sig []byte) (*Soc, error) {
 	s := New(id, ch)
-	s.signature = sig
 	o, err := NewOwner(owner)
```

Comment:
I think that `Owner` type is redundant and unnecessary. We could just settle for a simple byte slice and just verify the length in the constructor with
```go
	if len(address) != crypto.AddressSize {
		return nil, fmt.Errorf("soc: invalid owner address %x", address)
	}
```

WDYT?

Reply:
I agree. I'll do the change and remove the Owner type.